### PR TITLE
Add OpenStack Keystone User module

### LINKED ISSUE #283: keystone_user should use Keystone v3 (stable)
The OpenStack Identity API v3 is currently stable and keystone_user module should be upgraded to implement that version as the primary endpoint (falling back to v2.0).

[API v3](http://developer.openstack.org/api-ref-identity-v3.html)
[Unofficial Openstack modules for reference - no v2 there either](https://github.com/openstack-ansible/openstack-ansible-modules)

### PR DESCRIPTION
**Pull request by Shrews**
This is a replacement for PR #1598 and fixes #283

### CODE REVIEW

**Comment by Shrews** [comment]
cc: @j2sol and @emonty

**Comment by emonty** [comment]
Thanks Shrews - I think we should come back around and add some return value docs, but I'd be fine with that as a follow on.

**Comment by emonty** [comment]
shipit

File: cloud/openstack/os_user.py [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+    if module.params['password'] is not None:
+        return True
+
+    return False
+
+def main():
+
+    argument_spec = openstack_full_argument_spec(
+        name=dict(required=True),
+        password=dict(required=False, default=None),
```
Comment (omgjlk) [inline cloud/openstack/os_user.py:140]: This says password is not required, but the documentation string says that the password is required. Please reconcile.

**Comment by omgjlk** [comment]
This needs work, please don't ship it.

**Comment by emonty** [comment]
@j2sol has better eyeballs than I do

File: cloud/openstack/os_user.py (hunk 1 above)
Comment (Shrews) [inline cloud/openstack/os_user.py:140]: Fixed.

**Comment by emonty** [comment]
K. :+1:

**Comment by omgjlk** [comment]
shipit

**Comment by sivel** [comment]
# Merged

Hi!

This has been merged in, and will also be included in the next major release.

If you or anyone else has any further questions, please let us know by stopping by one of the two mailing lists, as appropriate:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Because this project is very active, we're unlikely to see comments made on closed tickets, but the mailing list is a great way to ask questions, or post if you don't think this particular
issue is resolved.

Thank you!

**Pull request merged**